### ISSUE TITLE
Don't report a 'backing property naming' error for internal properties

### ISSUE THREAD

**Issue by qwertyfinger** [pinterest/ktlint#3249]
## Expected Behavior

[Android Kotlin Style Guide - backing properties](https://developer.android.com/kotlin/style-guide#backing-properties)
[Kotlin Coding Conventions - backing properties](https://kotlinlang.org/docs/coding-conventions.html#names-for-backing-properties)

Don't report this backing property naming as an error. Sometimes I want to only expose a property for other components inside the module but still hide its implementation details. I don't see any hard rules forbidding this in Kotlin or Android style guides.

```kotlin
private val _actions = MutableSharedFlow<Action>()
internal val actions = _actions.asSharedFlow()
```

## Current Behavior

This results in an error:
```kotlin
private val _actions = MutableSharedFlow<Action>()
internal val actions = _actions.asSharedFlow()
```

Only this is allowed:
```kotlin
private val _actions = MutableSharedFlow<Action>()
val actions = _actions.asSharedFlow()
```

## Additional information
* Current version of ktlint: 1.8.0

**Comment by paul-dingemans** [comment]
You can either choose to [disable the rule](https://pinterest.github.io/ktlint/latest/faq/#how-do-i-enable-or-disable-a-rule) entirely, or to [suppress](https://pinterest.github.io/ktlint/latest/faq/#how-do-i-suppress-errors-for-a-lineblockfile) the case where you want to ignore the rule in that case.
